### PR TITLE
Fix Travis CI

### PR DESCRIPTION
**Pull request by ximion**
See issue https://github.com/dlang-community/D-YAML/issues/82 for details on this.

After working around one Travis bug and one Meson bug, I am now stuck with Meson not recognizing DMD as a valid compiler. My guess is that DMD changed the output of its `--version` command or some other signature that make Meson not recognize it as valid.

I am looking into the issue, so this PR is still WIP.

### CODE REVIEW

**Comment by ximion** [comment]
Okay, so Meson now has a PR to fix the DMD issue: https://github.com/mesonbuild/meson/pull/2889 - I am sure this fix will be merged soon, and then we'd need to wait for a bugfix release of Meson - which ideally will happen soon.
I don't want to add some hack to this project's CI to make this work meanwhile, as the proper fix is to make Meson recognize the latest DMD as valid compiler.

As soon as a new Meson release with this fix is out, Travis CI will pick it up automatically.
So, tl;dr, I think this PR is fine to merge for now, to fix most of the issues.

**Comment by BBasile** [comment]
Earlier today I' heard that TravisCI might include Meson, working out of the box, by default ,in their next platform release. You should check some times to times how is going the D-YAML CRON job, i was actually very surprised to see that  it has been broken since **4 months** (first only DMD 2.076 nightly then all the checks).

The problem is that you seem to be the only one to know the toolchain for including Meson.
At least thanks for handling this directly today.

**Comment by ximion** [comment]
I usually spot Meson bugs really quickly in my own projects. The issue here was that Travis itself changed behavior, which is something I can't actually fix. That issue was first spotted in Vibe, where it took me quite a while to come up with this workaround - and then I just assumed Travis would fix this bug rather quickly, as it was a behavior change breaking quite a large number of projects. It looks like they didn't do that, though.

For the DMD issue, I didn't spot that one because I don't have a project that is regularly building with DMD - so, my bad... On most Linux distros, people are also building with an optimizing compiler like LDC or GDC.
So yeah, to catch this kind of problem faster, I should subscribe to a project building with DMD regularly, or - maybe even better - have Meson's own CI test with DMD as well (that would require DMD to be packaged in Debian first, though - a task that is lacking manpower).

For the other Meson bug, I blame the Meson maintainers ;-)

**Comment by ximion** [comment]
The Meson issue is resolved in Meson master - we could use the Git snapshot here, but I think the better idea would be to just merge this PR and then wait for a new bugfix release of Meson.
I also have a PR for Meson to address the `subproject_dir` issue, that should be merged soonish, but that particular issue is worked around with this PR anyway.

**Comment by wilzbach** [comment]
Merging as I need this for https://github.com/dlang-community/D-YAML/pull/84
